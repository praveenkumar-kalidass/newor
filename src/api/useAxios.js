import axios from 'axios';

import useLoader from 'provider/Loader/useLoader';
import config from '../../config/config';

const useAxios = () => {
  const { addLoader, removeLoader } = useLoader();

  const instance = axios.create({
    baseURL: config.baseURL,
    timeout: 1000,
  });
  instance.interceptors.request.use((options) => {
    if (options.loader) {
      addLoader();
    }
    return options;
  }, (error) => (
    Promise.reject(error)
  ));
  instance.interceptors.response.use((response) => {
    if (response.config.loader) {
      removeLoader();
    }
    return response;
  }, (error) => {
    if (error.config.loader) {
      removeLoader();
    }
    return Promise.reject(error);
  });

  return instance;
};

export default useAxios;
