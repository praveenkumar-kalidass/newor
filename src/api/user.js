import axios from './index';
import config from '../../config/config';
import CONSTANT from '../constant';

export const login = ({ email, password }) => axios.post(`${config.baseURL}/api/user/v1/login`, {
  email,
  password,
  clientId: config.clientId,
  clientSecret: config.clientSecret,
  grantType: CONSTANT.AUTH_LITERAL.PASSWORD,
  responseType: CONSTANT.AUTH_LITERAL.CODE,
});

export const signup = (request) => axios.post(`${config.baseURL}/api/user/v1/signup`, request);
