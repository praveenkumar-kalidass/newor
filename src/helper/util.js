import CONSTANT from 'constant';

export const formatCurrency = (value) => {
  const formatter = new Intl.NumberFormat('hi-IN', { style: 'currency', currency: 'INR' });
  return formatter.format(value);
};

export const unformatCurrency = (value) => {
  const withoutSymbol = value.replace(CONSTANT.APP_LITERAL.RUPEE_SYMBOL, '');
  const currency = withoutSymbol.replace(/,/g, '');
  return parseFloat(currency);
};

export const formatUrlParams = (data) => Object.keys(data).map((key) => (
  `${encodeURIComponent(key)}=${encodeURIComponent(data[key])}`
)).join('&');
