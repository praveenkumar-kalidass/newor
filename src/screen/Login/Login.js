import React, {
  useState, useEffect, useMemo, useCallback,
} from 'react';
import { KeyboardAvoidingView, Platform } from 'react-native';
import {
  Flex, FormControl, useToast, Center, Image, Divider,
} from 'native-base';
import { useNavigation } from '@react-navigation/native';
import AsyncStorage from '@react-native-async-storage/async-storage';

import TitleBar from 'component/TitleBar/TitleBar';
import ToastAlert from 'component/ToastAlert';
import AppButton from 'component/AppButton';
import withBackground from 'helper/withBackground';
import { Background1 } from 'component/Background';
import NEWOR from 'asset/image/newor.png';
import CONSTANT from 'constant';
import validator from 'helper/validator';
import ROUTE from 'constant/route';
import useUserApi from 'api/useUser';
import useUser from 'provider/User/useUser';
import useTheme from 'theme/useTheme';
import useTranslation from 'translation/useTranslation';
import Translation from 'translation/Translation';
import usePreventBack from 'hook/usePreventBack';
import {
  INITIAL_STATE,
  FIELDS,
  SCHEMA,
} from './Login.schema';
import {
  FormContainer,
  InputField,
  Title,
  SignupLink,
} from './Login.style';

const Login = () => {
  const [fields, setFields] = useState(INITIAL_STATE);
  const [errorMessages, setErrorMessages] = useState(INITIAL_STATE);
  const [isSubmit, setIsSubmit] = useState(false);
  const navigation = useNavigation();
  const toast = useToast();
  const theme = useTheme();
  const { translate } = useTranslation();
  const { login } = useUserApi();
  const { setUser, isAuthorized } = useUser();
  usePreventBack();

  const handleFieldChange = (label, value) => {
    setFields({
      ...fields,
      [label]: value,
    });
  };

  const handleValidation = () => {
    setErrorMessages(validator(SCHEMA, fields, INITIAL_STATE));
  };

  const handleSubmit = () => {
    setErrorMessages(validator(SCHEMA, fields, INITIAL_STATE, true));
    setIsSubmit(true);
  };

  const doLogin = useCallback(async () => {
    try {
      const { data } = await login(fields);
      setIsSubmit(false);
      setUser(data.user);
      await AsyncStorage.setItem(CONSTANT.STORAGE_KEY.TOKEN, JSON.stringify({
        accessToken: data.accessToken,
        refreshToken: data.refreshToken,
      }));
      navigation.navigate(ROUTE.DASHBOARD_TAB);
    } catch (error) {
      setIsSubmit(false);
      let errorMessage = translate('ERROR_CODE.NEWOR_INTERNAL_SERVER_ERROR');
      const errorCode = error?.response?.data?.code;
      if (errorCode) {
        errorMessage = translate(`ERROR_CODE.${errorCode}`);
      }
      toast.show({
        render: () => <ToastAlert status="error" message={errorMessage} />,
        placement: 'top',
      });
    }
  }, [fields]);

  const isFormError = useMemo(
    () => Object.keys(errorMessages).some(
      (field) => !fields[field] || Boolean(errorMessages[field]),
    ),
    [errorMessages],
  );

  useEffect(() => {
    if (isSubmit && !isFormError) {
      doLogin();
    } else {
      setIsSubmit(false);
    }
  }, [navigation, isSubmit, isFormError]);

  return isAuthorized === false && (
    <KeyboardAvoidingView flex={1} behavior={Platform.OS === 'ios' ? 'padding' : 'height'}>
      <Flex flex={1} />
      <Flex flex={4} direction="row" p="5" justifyContent="center">
        <FormContainer>
          <TitleBar color={theme.color.SECONDARY_100} styleProps={{ mb: 5 }}>
            <Title>{CONSTANT.APP_NAME}</Title>
          </TitleBar>
          <Center><Image alt={CONSTANT.APP_NAME} source={NEWOR} size="xl" /></Center>
          <For each="field" index="index" of={FIELDS}>
            <FormControl key={`login-field-${index}`} isInvalid={errorMessages[field.key]}>
              <InputField
                testID={`login-input-${field.key}`}
                placeholder={translate(field.placeholder)}
                value={fields[field.key]}
                onChangeText={(value) => handleFieldChange(field.key, value)}
                onBlur={() => handleValidation()}
                type={
                  field.key === CONSTANT.AUTH_LITERAL.PASSWORD && CONSTANT.AUTH_LITERAL.PASSWORD
                }
                isInvalid={errorMessages[field.key]}
              />
              <FormControl.ErrorMessage>
                {errorMessages[field.key]}
              </FormControl.ErrorMessage>
            </FormControl>
          </For>
          <Translation
            tkey="LOGIN"
            as={AppButton}
            variant="primary"
            mb={5}
            testID="login-submit"
            onPress={handleSubmit}
            isDisabled={isFormError}
            isLoading={isSubmit}
          />
          <Flex direction="row">
            <Translation
              tkey="FORGOT_PASSWORD"
              as={SignupLink}
              link
              onPress={() => navigation.navigate(ROUTE.FORGOT_PASSWORD)}
            />
          </Flex>
          <Divider mt={5} mb={4} />
          <Flex direction="row" justifyContent="center">
            <Translation tkey="NEW_TO_NEWOR" as={SignupLink} />
            &nbsp;
            <Translation
              tkey="SIGNUP"
              as={SignupLink}
              link
              onPress={() => navigation.navigate(ROUTE.SIGNUP)}
            />
          </Flex>
        </FormContainer>
      </Flex>
      <Flex flex={1} />
    </KeyboardAvoidingView>
  );
};

export default withBackground({ Component: Login, Background: Background1 });
