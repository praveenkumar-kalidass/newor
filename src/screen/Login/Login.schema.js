import Joi from 'joi';

export const SCHEMA = Joi.object({
  email: Joi.string().email({ minDomainSegments: 2, tlds: { allow: ['com'] } }),
  password: Joi.string().alphanum().min(6).max(15)
    .required(),
});

export const INITIAL_STATE = {
  email: '',
  password: '',
};

export const FIELDS = [
  {
    key: 'email',
    placeholder: 'EMAIL',
  },
  {
    key: 'password',
    placeholder: 'PASSWORD',
  },
];
