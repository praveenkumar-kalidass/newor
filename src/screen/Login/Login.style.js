import styled from 'styled-components/native';
import {
  Box,
  Input,
  Text,
} from 'native-base';

export const FormContainer = styled(Box).attrs(({ theme }) => ({
  borderRadius: 20,
  flex: 1,
  alignSelf: 'center',
  p: 5,
  bg: theme.color.MODAL,
  elevation: 1,
  shadowColor: theme.color.BACKGROUND_0,
}))``;

export const InputField = styled(Input).attrs(({ isInvalid, theme }) => ({
  _focus: {
    bg: theme.color.BACKGROUND_100,
    borderColor: theme.color.SECONDARY_100,
  },
  mb: isInvalid ? 0 : 5,
}))`
  font-size: 14px;
`;

export const Title = styled(Text).attrs(() => ({
  bold: true,
  fontSize: 20,
}))`
  text-transform: capitalize;
`;

export const SignupLink = styled(Text).attrs(({ link, theme }) => ({
  fontSize: 'md',
  underline: link,
  color: link && theme.color.LINK,
}))``;
