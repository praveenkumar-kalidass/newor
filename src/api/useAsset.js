import useAxios from '.';
import config from '../../config';

const useAsset = () => {
  const axios = useAxios();

  const addDeposit = (request) => axios.post(`${config.baseURL}/api/deposit/v1`, request);

  const getAsset = (id) => axios.get(`${config.baseURL}/api/asset/v1/${id}`);

  return {
    addDeposit,
    getAsset,
  };
};

export default useAsset;
