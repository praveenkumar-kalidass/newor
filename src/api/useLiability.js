import useAxios from '.';
import config from '../../config';

const useLiability = () => {
  const axios = useAxios();

  const getLiability = (id) => axios.get(`${config.baseURL}/api/liability/v1/${id}`);

  const addLoan = (request) => axios.post(`${config.baseURL}/api/loan/v1`, request);

  return {
    getLiability,
    addLoan,
  };
};

export default useLiability;
