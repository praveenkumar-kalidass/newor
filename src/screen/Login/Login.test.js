import React from 'react';

import { login } from '../../api/user';
import { act, render, fireEvent } from '../../test/util';
import Login from './Login';

const mockToast = { show: jest.fn() };
jest.mock('native-base', () => {
  const nativeBase = jest.requireActual('native-base');
  return {
    ...nativeBase,
    useToast: () => mockToast,
  };
});
jest.mock('../../api/user', () => ({
  login: jest.fn(),
}));

describe('Login', () => {
  it('should match snapshot', async () => {
    const container = await render(<Login />);
    const { getByTestId } = container;
    await act(async () => {
      fireEvent(getByTestId('login-logo-container'), 'onLayout', {
        nativeEvent: {
          layout: {
            height: 300,
            width: 300,
          },
        },
      });
    });

    expect(container).toMatchSnapshot();
  });

  it('should show toast when login fails', async () => {
    login.mockRejectedValueOnce({ response: { data: { code: 'NEWOR_INTERNAL_SERVER_ERROR' } } });

    const { getByTestId } = await render(<Login />);

    await act(async () => {
      await fireEvent.changeText(getByTestId('login-input-email'), 'test@newor.com');
      await fireEvent(getByTestId('login-input-email'), 'onBlur');
      await fireEvent.changeText(getByTestId('login-input-password'), '123456');
      await fireEvent(getByTestId('login-input-password'), 'onBlur');
    });

    await act(async () => {
      await fireEvent.press(getByTestId('login-submit'));
    });

    expect(mockToast.show).toHaveBeenCalledTimes(1);
  });
});
