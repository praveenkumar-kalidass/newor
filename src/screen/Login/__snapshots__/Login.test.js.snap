// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "position": "absolute",
            },
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={1334}
          bbWidth={750}
          focusable={false}
          height={1334}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 1334,
                "width": 750,
              },
            ]
          }
          width={750}
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M0 200 L0 300 L750 200 L750 100 L0 200 Z"
              fill={4289527808}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M0 350 L0 370 L750 270 L750 250 L0 350 Z"
              fill={4289527808}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M0 1084 L0 1104 L750 1004 L750 984 L0 1084 Z"
              fill={4278199178}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M0 1154 L0 1254 L750 1154 L750 1054 L0 1154 Z"
              fill={4278199178}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        flex={1}
        onLayout={[Function]}
        style={
          Object {
            "paddingBottom": 0,
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        />
        <View
          dataSet={Object {}}
          style={
            Object {
              "display": "flex",
              "flex": 4,
              "flexDirection": "row",
              "justifyContent": "center",
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            }
          }
        >
          <View
            dataSet={Object {}}
            elevation={1}
            shadowColor="#000000"
            shadowOffset={
              Object {
                "height": 1,
                "width": 0,
              }
            }
            shadowOpacity={0.8}
            shadowRadius={1}
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "backgroundColor": "#eeeeee",
                  "borderRadius": 20,
                  "flex": 1,
                  "paddingBottom": 20,
                  "paddingLeft": 20,
                  "paddingRight": 20,
                  "paddingTop": 20,
                },
                Array [
                  Object {},
                ],
              ]
            }
          >
            <View
              dataSet={Object {}}
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                    "marginBottom": 20,
                  },
                  Array [
                    Object {},
                  ],
                ]
              }
              styleProps={
                Object {
                  "mb": 5,
                }
              }
            >
              <View
                dataSet={Object {}}
                style={
                  Array [
                    Object {
                      "backgroundColor": "#00238a",
                      "color": "#00238a",
                      "marginRight": 8,
                      "paddingRight": 8,
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
              />
              <Text
                dataSet={Object {}}
                style={
                  Array [
                    Object {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 20,
                      "fontStyle": "normal",
                      "fontWeight": "700",
                      "letterSpacing": 0,
                      "lineHeight": 30,
                      "textDecorationLine": undefined,
                    },
                    Array [
                      Object {
                        "textTransform": "capitalize",
                      },
                    ],
                  ]
                }
              >
                newor
              </Text>
            </View>
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                }
              }
            >
              <Image
                accessibilityLabel="newor"
                alt="newor"
                dataSet={Object {}}
                onError={[Function]}
                source={
                  Object {
                    "testUri": "../../../src/asset/image/newor.png",
                  }
                }
                style={
                  Object {
                    "height": 128,
                    "maxWidth": "100%",
                    "width": 128,
                  }
                }
              />
            </View>
            <View
              dataSet={Object {}}
              isInvalid=""
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <View
                dataSet={Object {}}
                isFocused={false}
                style={
                  Object {
                    "alignItems": "center",
                    "borderColor": "#d4d4d4",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "marginBottom": 20,
                    "overflow": "hidden",
                  }
                }
              >
                <TextInput
                  accessibilityDescribedBy="field-react-aria-1-feedback"
                  accessible={true}
                  dataSet={Object {}}
                  disabled={false}
                  editable={true}
                  isInvalid=""
                  isRequired={false}
                  nativeID="field-react-aria-1"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onKeyPress={[Function]}
                  placeholder="EMAIL"
                  placeholderTextColor="#a3a3a3"
                  readOnly={false}
                  required={false}
                  secureTextEntry={false}
                  selectionColor="#1f2937"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#171717",
                        "flex": 1,
                        "fontFamily": undefined,
                        "fontSize": 12,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "height": "100%",
                        "paddingBottom": 8,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "paddingTop": 8,
                        "width": "100%",
                      },
                      Array [
                        Object {
                          "fontSize": 14,
                        },
                      ],
                    ]
                  }
                  testID="login-input-email"
                  value=""
                />
              </View>
            </View>
            <View
              dataSet={Object {}}
              isInvalid=""
              style={
                Object {
                  "width": "100%",
                }
              }
            >
              <View
                dataSet={Object {}}
                isFocused={false}
                style={
                  Object {
                    "alignItems": "center",
                    "borderColor": "#d4d4d4",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "flexDirection": "row",
                    "marginBottom": 20,
                    "overflow": "hidden",
                  }
                }
              >
                <TextInput
                  accessibilityDescribedBy="field-react-aria-2-feedback"
                  accessible={true}
                  dataSet={Object {}}
                  disabled={false}
                  editable={true}
                  isInvalid=""
                  isRequired={false}
                  nativeID="field-react-aria-2"
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  onKeyPress={[Function]}
                  placeholder="PASSWORD"
                  placeholderTextColor="#a3a3a3"
                  readOnly={false}
                  required={false}
                  secureTextEntry={true}
                  selectionColor="#1f2937"
                  style={
                    Array [
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#171717",
                        "flex": 1,
                        "fontFamily": undefined,
                        "fontSize": 12,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "height": "100%",
                        "paddingBottom": 8,
                        "paddingLeft": 12,
                        "paddingRight": 12,
                        "paddingTop": 8,
                        "width": "100%",
                      },
                      Array [
                        Object {
                          "fontSize": 14,
                        },
                      ],
                    ]
                  }
                  testID="login-input-password"
                  value=""
                />
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessible={true}
              collapsable={false}
              colorScheme="primary"
              dataSet={Object {}}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#ad0000",
                  "borderRadius": 4,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "marginBottom": 20,
                  "opacity": 0.4,
                  "paddingBottom": 10,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 10,
                }
              }
              testID="login-submit"
            >
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                  }
                }
                test={true}
              >
                <View
                  dataSet={Object {}}
                  style={Object {}}
                >
                  <Text
                    dataSet={Object {}}
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "color": "#fafafa",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    LOGIN
                  </Text>
                </View>
              </View>
            </View>
            <View
              dataSet={Object {}}
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <Text
                dataSet={Object {}}
                link={true}
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": undefined,
                      "color": "#3498db",
                      "fontFamily": undefined,
                      "fontSize": 16,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 24,
                      "textDecorationLine": "underline",
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
              >
                FORGOT_PASSWORD
              </Text>
            </View>
            <View
              aria-orientation="horizontal"
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": "#d4d4d4",
                  "height": 1,
                  "marginBottom": 16,
                  "marginTop": 20,
                  "width": "100%",
                }
              }
              thickness="1"
            />
            <View
              dataSet={Object {}}
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <Text
                dataSet={Object {}}
                style={
                  Array [
                    Object {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 16,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 24,
                      "textDecorationLine": undefined,
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
              >
                NEW_TO_NEWOR
              </Text>
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 14,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 21,
                    "textDecorationLine": undefined,
                  }
                }
              >
                 
              </Text>
              <Text
                dataSet={Object {}}
                link={true}
                onPress={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": undefined,
                      "color": "#3498db",
                      "fontFamily": undefined,
                      "fontSize": 16,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 24,
                      "textDecorationLine": "underline",
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
              >
                SIGNUP
              </Text>
            </View>
          </View>
        </View>
        <View
          dataSet={Object {}}
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        />
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      elevation={0}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": -600,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 600,
          },
        ]
      }
    >
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {
              "backgroundColor": "#eeeeee",
              "flex": 1,
              "justifyContent": "space-between",
              "paddingBottom": 80,
              "paddingTop": 80,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Array [
              Object {},
              Object {
                "dataSet": Object {},
              },
            ]
          }
          data={
            Array [
              Object {
                "action": "press",
                "icon": <Icon
                  allowFontScaling={false}
                  color="#000000"
                  name="power-off"
                  size={24}
                />,
                "id": "LOGOUT",
                "label": "LOGOUT",
              },
            ]
          }
          dataSet={Object {}}
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={Object {}}
          viewabilityConfigCallbackPairs={Array []}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  dataSet={Object {}}
                  style={
                    Array [
                      Object {
                        "borderBottomWidth": 1,
                        "borderColor": "#7f7f7f",
                        "paddingBottom": 8,
                        "paddingLeft": 16,
                        "paddingRight": 20,
                        "paddingTop": 8,
                      },
                      Array [
                        Object {},
                      ],
                    ]
                  }
                >
                  <View
                    dataSet={Object {}}
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 24,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      
                    </Text>
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "width": 12,
                        }
                      }
                    />
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        LOGOUT
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
